google/cloud/bigtable/table_admin_legacy_test.cc

```diff
@@ -95,159 +95,190 @@ class TableAdminTest : public ::testing::Test {
+    return [this, backup_id](
+               grpc::ClientContext* context,
+               ::google::iam::v1::SetIamPolicyRequest const& request,
+               ::google::iam::v1::Policy* response) {
+      EXPECT_STATUS_OK(IsContextMDValid(
+          *context, "google.bigtable.admin.v2.BigtableTableAdmin.SetIamPolicy",
+          backup_id));
+      EXPECT_NE(nullptr, response);
+      *response = request.policy();
+      return grpc::Status::OK;
     };
+  }
 
-/**
- * Helper class to create the expectations for a simple RPC call.
- *
- * Given the type of the request and responses, this struct provides a function
- * to create a mock implementation with the right signature and checks.
- *
- * @tparam RequestType the protobuf type for the request.
- * @tparam ResponseType the protobuf type for the response.
- */
-template <typename RequestType, typename ResponseType>
-struct MockRpcFactory {
-  using SignatureType = grpc::Status(grpc::ClientContext* ctx,
-                                     RequestType const& request,
-                                     ResponseType* response);
-
-  /// Refactor the boilerplate common to most tests.
-  static std::function<SignatureType> Create(
-      std::string const& expected_request, std::string const& method) {
-    return std::function<SignatureType>(
-        [expected_request, method](grpc::ClientContext* context,
-                                   RequestType const& request,
-                                   ResponseType* response) {
-          EXPECT_STATUS_OK(IsContextMDValid(
-              *context, method, google::cloud::internal::ApiClientHeader()));
-          if (response == nullptr) {
-            return grpc::Status(grpc::StatusCode::INVALID_ARGUMENT,
-                                "invalid call to MockRpcFactory::Create()");
-          }
-          RequestType expected;
-          // Cannot use ASSERT_TRUE() here, it has an embedded "return;"
-          EXPECT_TRUE(google::protobuf::TextFormat::ParseFromString(
-              expected_request, &expected));
-          EXPECT_THAT(expected, IsProtoEqual(request));
-
-          return grpc::Status::OK;
-        });
+  std::function<grpc::Status(grpc::ClientContext*,
+                             btadmin::ListBackupsRequest const&,
+                             btadmin::ListBackupsResponse*)>
+  CreateListBackupsMock(std::string const& expected_token,
+                        std::string const& returned_token,
+                        std::vector<std::string> const& backup_names) {
+    return [this, expected_token, returned_token, backup_names](
+               grpc::ClientContext* context,
+               btadmin::ListBackupsRequest const& request,
+               btadmin::ListBackupsResponse* response) {
+      EXPECT_STATUS_OK(IsContextMDValid(
+          *context, "google.bigtable.admin.v2.BigtableTableAdmin.ListBackups",
+          google::cloud::internal::ApiClientHeader()));
```

Comment:
This is setting the `resource_name` to `google::cloud::internal::ApiClientHeader()`... I'm guessing you don't want that.

Reply:
Fixed.